### DISCUSSION TITLE
docker: failed to register layer: error creating overlay mount to ${path}: too many levels of symbolic linksls

### DISCUSSION THREAD

**Discussion by streetycat** [buckyos/CYFS#243] [Q&A] [answered]
I meet a error about docker:

There are some error logs in `app-manager`:

```

app 9tGpLNniojkX6rVCdDi7t6BK1sMqDjw32RFhxQUReiFH run install cmd npm i --production --force, exit code 125

docker install failed. app:9tGpLNniojkX6rVCdDi7t6BK1sMqDjw32RFhxQUReiFH failed, err: (Failed, , None)
[app-manager_1628_rCURRENT.log](https://github.com/buckyos/CYFS/files/11342016/app-manager_1628_rCURRENT.log)

```

I found some logs before the errors as follow:
```
will spawn cmd: docker run --name decapp-9tgplnniojkx6rvcddi7t6bk1smqdjw32rfhxqureifh-install --cap-add NET_ADMIN --cap-add NET_RAW --rm --init --network bridge -v /cyfs/log/app/9tGpLNniojkX6rVCdDi7t6BK1sMqDjw32RFhxQUReiFH:/cyfs/log/app -v /cyfs/data/app/9tGpLNniojkX6rVCdDi7t6BK1sMqDjw32RFhxQUReiFH:/cyfs/data/app/9tGpLNniojkX6rVCdDi7t6BK1sMqDjw32RFhxQUReiFH -v /cyfs/app/dockerfile/9tGpLNniojkX6rVCdDi7t6BK1sMqDjw32RFhxQUReiFH/start.sh:/opt/start.sh:ro -v /cyfs/tmp:/cyfs/tmp -v /etc/localtime:/etc/localtime:ro -v /run/systemd/resolve/resolv.conf:/etc/resolv.conf:ro -v /cyfs/app/9tGpLNniojkX6rVCdDi7t6BK1sMqDjw32RFhxQUReiFH:/opt/app buckyos/dec-app-base:v1 npm i --production --force
```

I execute the command in the shell of the `OOD`, it shows error as follow:
```
docker: failed to register layer: error creating overlay mount to /var/lib/docker/overlay2/216713d197b9f8e040b4f7c77ceaafb43139feeb8fc037caf7e155436b2dd800/merged: too many levels of symbolic links.
See 'docker run --help'.
```

I has report it in this [issue](https://github.com/buckyos/CYFS/issues/216#issuecomment-1514495958). It said that it's a problem with the docker environment.

But I also tried to run other images as follow:

```
docker run wordpress:apache
```

It is running success:

```
CONTAINER ID   IMAGE              COMMAND                  CREATED          STATUS          PORTS     NAMES
b5b5f509dce6   wordpress:apache   "docker-entrypoint.s…"   45 minutes ago   Up 43 minutes   80/tcp    magical_hellman
```

Therefore, I guess it may also be related to our mirroring or usage, maybe it's a compatibility issue.

And it worked fine at the beginning of the year, is there some update caused it?

**Comment by streetycat** [comment, marked as answer]
I have resolved it with reinstall the docker.

But I don't known what happened. Thankfully, the data is preserved without lost.

**Comment by lurenpluto** [comment]
This problem has also been mentioned before, I googled the problem, it seems to be in the docker inside the creation of the wrong soft link, should be app-manager some place operation problems, may need to review the relevant code and logic that may be involved

There are also some more specific diagnostic methods to see which links are wrong, next time if these problems still occur, I hope to help diagnose the problem to locate
